eng/Directory.Build.Common.props

```diff
@@ -1,4 +1,21 @@
 ﻿<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <!-- Setup project type properties -->
+  <PropertyGroup>
+    <IsMgmtLibrary Condition="'$(IsMgmtLibrary)' == '' and $(MSBuildProjectName.StartsWith('Azure.ResourceManager.')))">true</IsMgmtLibrary>
+
+    <!-- Client in this context is any library that is following the new Azure SDK guidelines -->
+    <IsClientLibrary Condition="'$(IsClientLibrary)' == '' and $(MSBuildProjectName.StartsWith('Azure.'))">true</IsClientLibrary>
+
+    <IsTestProject Condition="$(MSBuildProjectName.EndsWith('.Tests'))">true</IsTestProject>
```

Comment:
I think all these should check if they are already set. IIRC, I use `IsSamplesProject` in sdk/keyvault/samples, or maybe I used something similar because I couldn't override this. Effectively, I want sample-like behavior.

Reply:
I will add a condition to see if they are already set.